src/result.js

```diff
@@ -45,6 +48,45 @@ class Result {
     return insertIntoIndexHTML(this._html, this._head, this._body, this._bodyAttributes);
   }
 
+  /**
+   * Returns the HTML representation of the rendered route, inserted
+   * into the application's `index.html`, split into chunks.
+   * The first chunk contains the document's head, the second contains the body
+   * until just before the shoebox tags (if there are any) and the last chunk
+   * contains the shoebox tags and the closing `body` tag. If there are no
+   * shoebox tags, there are only 2 chunks and the second one contains the
+   * complete document body, including the closing `body` tag.
+   *
+   * @returns {Promise<Array<String>>} the application's DOM serialized to HTML, split into chunks
+   */
+  chunks() {
+    return insertIntoIndexHTML(this._html, this._head, this._body, this._bodyAttributes).then((html) => {
```

Comment:
A minor optimization that can be done here is:

Instead of replacing the entire index.html, you may want to think about only replacing the chunk lazily. This will enable sending the head faster. Can be done at a later point too. I think with streaming you want to make sure you do an early flush as soon as possible.

Reply:
Yeah, ideally we could stream the head as soon as it is rendered, even while the rest of the document is still being rendered. My understanding is that that is currently not possible though as we have to wait until `visit` resolves at which point the complete document has been rendered already? Also I guess in order for that to work we'd have to make this a generator function rather than returning a promise?

Or maybe you're referring to something else that I don't get :)

Reply:
Yeah I wasn't refering to waiting for `visit`. But right now we parsing the entire `index.html` and replacing body as well when we only want to send the head first.

Reply:
Yeah, good point. But I think we'd have to turn this into a generator function then though to be able to give control back to the caller once the first chunk has been calculated. That would be a breaking change though…